File: Ruby Free/demos/browser/features/step_definitions/api_test_steps.rb
Given(/^I search for news articles that reference '(.*)'$/) do |search_term|
  @search_term = search_term
  @results = Api.find_articles(search_term: @search_term)
end

Then(/^I receive results that are relevant to my search term$/) do
  if @results
    Api.example_data_outputs_from_response(@results)
    Api.assert_results_match(@results, @search_term)
  end
end

File: Typescript/demos/browser/features/step_definitions/apiTestSteps.ts
import {Given, Then} from "@cucumber/cucumber";
import * as apiHelper from '../support/api/apiHelper';

Given(/^I search for news articles that reference '(.*)'$/, async function (data) {
  this.searchTerm = data
  this.results = await apiHelper.findArticles(this.searchTerm);
});

Then(/^I receive results that are relevant to my search term$/, async function () {
  apiHelper.exampleDataOutputsFromResponse(this.results);
  apiHelper.assertResultsMatch(this.results, this.searchTerm);
});
